service/controller/resource/app/current.go

```diff
@@ -51,8 +52,16 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) ([]*v1a
 	{
 		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding apps in tenant cluster %#q", clusterConfig.ID))
 
+		selectorLabels := []string{
+			fmt.Sprintf("%s=%s", label.ManagedBy, project.Name()),
+		}
+
+		if r.provider == "azure" {
+			selectorLabels = append(selectorLabels, "app!=nginx-ingress-controller")
+		}
+
 		o := metav1.ListOptions{
-			LabelSelector: fmt.Sprintf("%s=%s", label.ManagedBy, project.Name()),
+			LabelSelector: strings.Join(selectorLabels, ","),
```

Comment:
@rossf7 @whites11 sorry for the noise in rest of the PR (coming from mandatory upgrade of architect-orb), this selected part is the essence enabling making NGINX optional on Azure.

Second important part of the solution is in [Azure 12.0.0 platform release definition](https://github.com/giantswarm/releases/pull/343/files) - that PR has also noise due to sync with refactoring on releases repo master. Anyway, key part in Azure 12.0.0 release definition is use of this cluster-operator and dropping nginx app from list of the pre-installed apps.

New cluster-operator ignores existing nginx IC App CR, so current and desired App CRs will match.

I've tested this on ghost and it works, with some unwanted side-effects. I need your help to evaluate the approach, and unwanted side-effects if they and proposed workaround are acceptable, or you have idea how to further improve it or .

Older Azure platform releases use older cluster-operator, and they do not ignore nginx App CR.
As of this cluster-operator we start ignoring nginx App CR, using that negation selector label.
This enables:
- Azure 12.0.0 release to remove nginx App CR from list of components
- When existing clusters get upgraded to 12.0.0, existing nginx App CR remains to run uninterrupted
- Customer can perform cluster upgrade with no special assistance from SEs
- Future Azure 12+ releases can copy 12.0.0 definition as template and modify what they need to modify, without any special additional cleanup steps due to removing nginx, future releases carry over essential logic by using same cluster-operator
- Node pools Azure release if it makes use of cluster-operator master, doesn't have to 

managed-by cluster-operator label is not automatically removed. Until label is removed, there are unwanted side-effects:
- Upon cluster upgrade, nginx IC App doesn't get listed in Happa - neither as pre-installed nor as installed optional
- Left over label on nginx IC App CR might be confusing to the uninitiated

When it comes to left-over managed-by cluster-operator label cleanup, I'm reluctant to modify/pollute app-operator with that logic. If we don't come up with another automatic solution, maybe SEs can help with the cleanup of the managed-by label, post cluster upgrades. Maybe we can come up with business hours notification alert about such nginx App CRs with left-over label.

Additional dilemma is should we keep having external-dns pre-installed. There are pros and cons to it. IMO we should keep it for now, can deal with the problem holistically in a followup project.

WDYT?

Reply:
I like this approach. Very pragmatic.
I don't agree much with the manual cleanup of the label leftover. 
I would honestly add the cleanup logic to the cluster operator code, and remove it from a future release. But that's my point of view, the manual approach would work as well and the side effects are not impacting that much.

As of external-dns I'm ok making it optional but it is better to leave that out from this release IMHO, because it is already huge.

Reply:
Yes I like the approach but I agree with @whites11. Let's tackle external-dns in a later release. Since IC is a complex app on its own.

I'd also prefer an automated approach. As some TC upgrades are out of hours and manual steps are easy to miss. This will add some complexity to the code but this is in the legacy branch so I'm fine with that.

Reply:
> add the cleanup logic to the cluster operator code

I wanted to, but don't know how to manage App CRs twice in same operator - one existing logic with modification to ignore nginx App CRs, and separate one to remove the managed-by label from nginx App CRs only.

Reply:
> I wanted to, but don't know how to manage App CRs twice in same operator

Hmm its a bit complex but you could do it via a migration resource. So you keep this logic to ignore nginx App CRs.

In the migration resource once we're happy the TC upgrade is done it removes the managed-by label. There are already 2 migration resources.

- appmigration is for replacing the chartconfig CR with an app CR
- configmapmigration copies user CMs to the new location in the CP

WDYT? Does that help?